### PR TITLE
Strict types

### PR DESCRIPTION
**Pull request by canadianveggie**
Follow up on #136 
Turn on `noImplicitAny` and `strictNullChecks`.

The only big code change was to Knowledge which no longer allows partial definitions. Cleans up the code and makes the types simpler, but need to ensure we handle backward compatibility with saved state.

Also changed scripts/emojichec.js to download the file instead of relying on a local (but not checked in) version.

We do have a big typeschecking hole in index.html, but otherwise things look great.

### CODE REVIEW

File: scripts/emojicheck.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 // @ts-check
 
+const https = require("https");
```
Comment (wachunga) [inline scripts/emojicheck.js:3]: we use axios in scripts/dictionary — probably cleaner to use it here, too

File: tsconfig.json [hunk 2]
```diff
@@ -1,30 +1,24 @@
...
+    "declaration": false /* Generate .d.ts files from TypeScript and JavaScript files in your project. */,
+    "outDir": "./dist" /* Specify an output folder for all emitted files. */,
+    "forceConsistentCasingInFileNames": true /* Ensure that casing is correct in imports. */,
+
+    "strict": true,
     // TODO: enable below
-    "noImplicitAny": false,                           
-    "strictNullChecks": false,
+    "noImplicitAny": true,
+    "strictNullChecks": true
```
Comment (wachunga) [inline tsconfig.json:22]: Thanks for turning these on. I don't think there's anything stopping us from bundling with TS as a next step (future PR)... unless I'm forgetting something about alpine?

**Review by wachunga — APPROVED** [review]

File: scripts/emojicheck.js (hunk 1 above)
Comment (canadianveggie) [reply scripts/emojicheck.js:3]: Nice. I didn't want to add a dependency, but axios is a lot cleaner.

File: tsconfig.json (hunk 2 above)
Comment (canadianveggie) [reply tsconfig.json:22]: A bundler would be nice. But only if it's easy.

**Pull request merged**